packages/client/src/grpc-retry.ts

```diff
@@ -0,0 +1,86 @@
+import {
+  InterceptingCall,
+  Interceptor,
+  ListenerBuilder,
+  Metadata,
+  RequesterBuilder,
+  StatusObject,
+} from '@grpc/grpc-js';
+import * as grpc from '@grpc/grpc-js';
+
+const maxRetries = 10;
```

Comment:
I'd rather avoid hardcoding this here, it'd be nicer if this was configurable with a sane default.
We don't have to expose this knob to users, just would make it easier to test.